oximeter/db/src/oxql/point.rs

```diff
@@ -0,0 +1,1725 @@
+            anyhow::bail!(
+                "Cannot access value array as integer \
+                distribution type, it has type {}",
+                self.data_type(),
+            );
+        };
+        Ok(inner)
+    }
+
+    // Access the inner array of integer distribution, if possible.
+    pub(super) fn as_integer_distribution_mut(
+        &mut self,
+    ) -> Result<&mut Vec<Option<Distribution<i64>>>, Error> {
+        let ValueArray::IntegerDistribution(inner) = self else {
+            anyhow::bail!(
+                "Cannot access value array as integer \
+                distribution type, it has type {}",
+                self.data_type(),
+            );
+        };
+        Ok(inner)
+    }
+
+    /// Access the values as an array of double distribution, if they have that
+    /// type.
+    pub fn as_double_distribution(
+        &self,
+    ) -> Result<&Vec<Option<Distribution<f64>>>, Error> {
+        let ValueArray::DoubleDistribution(inner) = self else {
+            anyhow::bail!(
+                "Cannot access value array as double \
+                distribution type, it has type {}",
+                self.data_type(),
+            );
+        };
+        Ok(inner)
+    }
+
+    // Access the inner array of double distributions, if possible.
+    pub(super) fn as_double_distribution_mut(
+        &mut self,
+    ) -> Result<&mut Vec<Option<Distribution<f64>>>, Error> {
+        let ValueArray::DoubleDistribution(inner) = self else {
+            anyhow::bail!(
+                "Cannot access value array as double \
+                distribution type, it has type {}",
+                self.data_type(),
+            );
+        };
+        Ok(inner)
+    }
+
+    fn push_missing(&mut self, datum_type: DatumType) -> Result<(), Error> {
+        match datum_type {
+            DatumType::Bool => self.as_boolean_mut()?.push(None),
+            DatumType::I8
+            | DatumType::U8
+            | DatumType::I16
+            | DatumType::U16
+            | DatumType::I32
+            | DatumType::U32
+            | DatumType::I64
+            | DatumType::U64
+            | DatumType::CumulativeI64
+            | DatumType::CumulativeU64 => self.as_integer_mut()?.push(None),
+            DatumType::F32
+            | DatumType::F64
+            | DatumType::CumulativeF32
+            | DatumType::CumulativeF64 => self.as_double_mut()?.push(None),
+            DatumType::String => self.as_string_mut()?.push(None),
+            DatumType::Bytes => {
+                anyhow::bail!("Bytes data types are not yet supported")
+            }
+            DatumType::HistogramI8
+            | DatumType::HistogramU8
+            | DatumType::HistogramI16
+            | DatumType::HistogramU16
+            | DatumType::HistogramI32
+            | DatumType::HistogramU32
+            | DatumType::HistogramI64
+            | DatumType::HistogramU64
+            | DatumType::HistogramF32
```

Comment:
Should `HistogramF32` and `HistogramF64` map to `self.as_double_distribution_mut`?

Reply:
Indeed, nice catch.

Reply:
Moved in aefa41886, and added tests.